File: trabalho3/Sistema py/views/menu.py
"""Tela 0 — Menu principal do sistema."""

import tkinter as tk

from config import CORES, FONTES
from components.widgets import botao_menu


class MenuFrame(tk.Frame):
    """Tela inicial com acesso às duas funcionalidades do protótipo: cadastrar
    um lote de produto e consultar os lotes disponíveis."""

    def __init__(self, parent, app):
        super().__init__(parent, bg=CORES["fundo"])
        self.app = app

        conteudo = tk.Frame(self, bg=CORES["fundo"])
        conteudo.place(relx=0.5, rely=0.42, anchor="center")

        tk.Label(
            conteudo, text="SOS Abobrinha", bg=CORES["fundo"],
            fg=CORES["verde_principal"], font=(FONTES["titulo"][0], 20, "bold"),
        ).pack(pady=(0, 6))

        tk.Label(
            conteudo, text="Redistribuição de excedentes agrícolas",
            bg=CORES["fundo"], fg=CORES["texto_secundario"], font=FONTES["subtitulo"],
        ).pack(pady=(0, 30))

        opcoes = tk.Frame(conteudo, bg=CORES["fundo"])
        opcoes.pack(fill="x")

        botao_menu(
            opcoes, "Cadastrar lote de produto",
            "Registrar novo excedente agrícola",
            lambda: self.app.show_frame("passo1"),
        ).pack(fill="x", pady=(0, 12))

        botao_menu(
            opcoes, "Consultar lotes disponíveis",
            "Ver excedentes disponíveis para solicitação",
            lambda: self.app.show_frame("consulta"),
        ).pack(fill="x")

File: trabalho3/Sistema JS/js/views/menu.js
// ── TELA 0: Menu principal ──

/**
 * Monta a estrutura HTML da tela de menu e a insere em #app.
 */
function construirTelaMenu() {
  const app = document.getElementById("app");

  const screen = document.createElement("div");
  screen.id = "screen-menu";
  screen.className = "screen";

  screen.innerHTML = `
    <div class="app-header">
      <p class="app-titulo">SOS Abobrinha</p>
      <p class="app-subtitulo">Redistribuição de excedentes agrícolas</p>
    </div>

    <button class="btn-menu" id="btn-ir-cadastro">
      <div class="icone">📦</div>
      <div>
        <div class="titulo-btn">Cadastrar lote de produto</div>
        <div class="desc-btn">Registrar novo excedente agrícola</div>
      </div>
      <span class="seta">›</span>
    </button>

    <button class="btn-menu" id="btn-ir-consulta">
      <div class="icone">🔍</div>
      <div>
        <div class="titulo-btn">Consultar lotes disponíveis</div>
        <div class="desc-btn">Ver excedentes disponíveis para solicitação</div>
      </div>
      <span class="seta">›</span>
    </button>
  `;

  app.appendChild(screen);

  screen.querySelector("#btn-ir-cadastro").addEventListener("click", () => {
    App.navegarPara("passo1");
  });
  screen.querySelector("#btn-ir-consulta").addEventListener("click", () => {
    App.navegarPara("consulta");
  });
}

construirTelaMenu();

App.registrarTela("menu", {});
